sharktank/sharktank/utils/benchmark.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2025 Advanced Micro Devices, Inc.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+from typing import Any
+import iree.runtime
+import subprocess
+import json
+import sys
+import pandas
+from pathlib import Path
+import os
+from os import PathLike
+
+
+def _run_program(
+    args: tuple[str],
+):
+    process_result = subprocess.run(
+        args=args,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    )
+
+    out = process_result.stdout.decode()
+    err = process_result.stderr.decode()
+
+    if process_result.returncode != 0:
+        raise RuntimeError(f"stderr:\n{err}\nstdout:\n{out}")
+
+    if err != "":
+        print(err, file=sys.stderr)
```

Comment:
We have other test suites that log the compile/run/benchmark commands before issuing them. I don't see that done here, but maybe it should be. Otherwise, these error messages would be missing some context.

You could also use `logging` instead of `print()` and use debug/info/warning/error levels if you expect that logging to be noisy enough that a developer might want to silence some parts of it.

Reply:
I improved logging and the exception message.